sdk/core/azure-core/tests/perf_tests/upload_binary.py

```diff
@@ -0,0 +1,80 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import time
+from wsgiref.handlers import format_date_time
+from devtools_testutils.perfstress_tests import RandomStream, AsyncRandomStream
+
+from azure.core.rest import HttpRequest
+from azure.core.exceptions import (
+    HttpResponseError,
+    map_error,
+)
+from ._test_base import _BlobTest
+
+
+class UploadBinaryDataTest(_BlobTest):
+    def __init__(self, arguments):
+        super().__init__(arguments)
+        blob_name = "uploadtest"
+        self.blob_endpoint = f"{self.account_endpoint}{self.container_name}/{blob_name}"
+        self.upload_stream = RandomStream(self.args.size)
+        self.upload_stream_async = AsyncRandomStream(self.args.size)
+
+    def run_sync(self):
+        self.upload_stream.reset()
+        data = self.upload_stream.read(self.args.size)
```

Comment:
You shouldn't need to read the data first - does it not work if you just pass the stream in?

Reply:
When I pass `content=self.upload_stream` directly in the HttpRequest without reading, it returns this auth error for some reason:
```
Error: Operation returned an invalid status 'Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.'
Content: <?xml version="1.0" encoding="utf-8"?><Error><Code>AuthenticationFailed</Code><Message>Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
RequestId:e09f37f1-901e-000d-7a97-4ebbff000000
Time:2024-01-24T07:30:18.2904387Z</Message><AuthenticationErrorDetail>The MAC signature found in the HTTP request is not the same as any computed signature. 
```
Not sure why this would change the auth but will compare signatures.

Reply:
OOB: issue with requests transport stream - size is 1 less than was specified

Reply:
issue resolved - description here: #33997